### PR TITLE
NP-7 // Provide new configuration option

### PR DESCRIPTION
**Pull request by aliya2409**

### CODE REVIEW

File: src/main/java/com/javalab/newsportal/config/WebAppInitializer.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        AnnotationConfigWebApplicationContext context
+                = new AnnotationConfigWebApplicationContext();
+        context.setConfigLocation("com.javalab.newsportal.config");
+        container.addListener(new ContextLoaderListener(context));
+        ServletRegistration.Dynamic dispatcher = container
+                .addServlet("dispatcher", new DispatcherServlet(context));
+
+        dispatcher.setLoadOnStartup(1);
+        dispatcher.addMapping("/");
+    }
```
Comment (cancion2able) [inline src/main/java/com/javalab/newsportal/config/WebAppInitializer.java]: This config can be simplified by extending AbstractAnnotationConfigDispatcherServletInitializer

File: src/main/java/com/javalab/newsportal/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    public News news() {
+        return new News();
+    }
+
+    @Bean
+    public Comment comment() {
+        return new Comment();
+    }
+
+    @Bean
```
Comment (cancion2able) [inline src/main/java/com/javalab/newsportal/config/AppConfig.java]: From here we can rely on component scanning (by marking our DAOs as @Repository and services as @Service). No need to manually create them, let the dependency injection come into play.

File: src/main/java/com/javalab/newsportal/service/NewsRetrievalServiceImpl.java [hunk 3]
```diff
@@ -14,6 +15,7 @@ public NewsRetrievalServiceImpl(NewsDAO newsDAO) {
 
     @Override
     public News retrieve(Long id) {
-        return newsDAO.findById(id);
+        return newsDAO.findById(id).orElseThrow(
+                IllegalArgumentException::new);
```
Comment (cancion2able) [inline src/main/java/com/javalab/newsportal/service/NewsRetrievalServiceImpl.java]: Let's not wait and create custom exception. Also provide a concrete message here.

File: src/main/java/com/javalab/newsportal/service/comments/CommentsRetrievalServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                .sorted(commentDateComparator)
+                .collect(Collectors.toList());
+    }
+
+    private static <T> Predicate<T> andLogFilteredOutValues(Predicate<T> predicate) {
+        return value -> {
+            if (predicate.test(value)) {
+                return true;
+            } else {
+                System.out.println("Filtered: " + value);
```
Comment (cancion2able) [inline src/main/java/com/javalab/newsportal/service/comments/CommentsRetrievalServiceImpl.java]: It's time to add a decent logger to let the world know which services are behaving incorrectly.

File: src/main/java/com/javalab/newsportal/service/comments/CommentsRetrievalServiceImpl.java [hunk 5]
```diff
@@ -21,12 +22,16 @@ public CommentsRetrievalServiceImpl(CommentDAO commentDAO) {
...
-        Comparator<Comment> commentRatingComparator = Comparator.comparing(Comment::getRating, Comparator.reverseOrder());
+    public List<Comment> retrieve(News news, String sortBy) {
+        Comparator<Comment> comparator;
+        if (Constants.RATING_SORTING_OPTION.equals(sortBy)) {
+            comparator = Comparator.comparing(Comment::getRating, Comparator.reverseOrder());
+        } else {
+           comparator = Comparator.comparing(Comment::getCreationDate, Comparator.reverseOrder());
+        }
         return commentDAO.findByNews(news).stream()
                 .filter(andLogFilteredOutValues(p -> !p.getContent().toLowerCase().contains(PROHIBITED_WORD)))
```
Comment (cancion2able) [inline src/main/java/com/javalab/newsportal/service/comments/CommentsRetrievalServiceImpl.java]: Try taking a look at Stream API peek() method and think how this implementation can be improved

Comment (aliya2409) [reply src/main/java/com/javalab/newsportal/service/comments/CommentsRetrievalServiceImpl.java]: I have forgotten about this one. I'll do it tommorow.

**Pull request merged**